misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java

```diff
@@ -319,86 +236,163 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
                 if (!shouldSkip.get()) {
                     suite.addTests(test);
                 }
+            } else if (MOCK_ANNOTATION_NAME.equals(annotationName)) {
+                String[] vals = new String[2];
+                // TODO: when default values are supported in annotation struct we can remove this
+                vals[0] = packageName;
+                if (attachmentNode.getExpression() instanceof BLangRecordLiteral) {
```

Comment:
Let's use the node Kind to check the condition.
i.e. `attachmentNode.getExpression().getKind() == NodeKind.RECORD_LITERAL_EXP`